pkg/client/connection.go

```diff
@@ -29,5 +29,8 @@ type Connection interface {
 	Connect() error
 	// Close closes the connection, releasing all the resources that it uses. Once closed the
 	// connection can't be reused.
-	Close() error
+	Disconnect() error
+
+	Publish(m Message) error
+	Subscribe() error
```

Comment:
Publish and subscribe to what topic? That, the topic name, as well as other properties, should either be part of the signature of the `Publish` method, or else there should be a `Publisher` interface that handles that.